synapse/events/spamcheck.py

```diff
@@ -307,6 +304,16 @@ async def check_event_for_spam(
                 else:
                     # This spam-checker rejects the event either with a `str`
                     # or with a `Codes`. In either case, we stop here.
+                    if not isinstance(res, str):
```

Comment:
If the comment is true, the condition is `False`.

I'm not quite clear what the aim of the condition is, but if it's a defense against the module returning an invalid value, then we want something like:

```suggestion
                elif not isinstance(res, str):
                    logger.warning("Module returned invalid value, rejecting message as spam")
                    res = "This message has been rejected as probable spam"
                else:
                    # This spam-checker rejects the event either with a `Codes`
                    # or a different `str`. In either case, we stop here.
                    pass
```